src/vscode-dts/vscode.proposed.badges.d.ts

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	// https://github.com/microsoft/vscode/issues/62783 @matthewjamesadam
+
+	/**
+	 * A badge presenting a number
+	 */
+	export interface NumberBadge {
+
+		/**
+		 * The number to present
+		 */
+		number: number;
+
+		/**
+		 * A label to present in tooltips
+		 */
+		label: string;
+	}
+
+	/**
+	 * A badge presenting text
+	 */
+	export interface TextBadge {
+
+		/**
+		 * The text to present
+		 */
+		text: string;
+
+		/**
+		 * A label to present in tooltips
+		 */
+		label: string;
+	}
+
+	/**
+	 * A badge presenting an icon
+	 */
+	export interface IconBadge {
+
+		/**
+		 * The icon to present
+		 */
+		icon: ThemeIcon;
+
+		/**
+		 * A label to present in tooltips
+		 */
+		label: string;
+	}
+
+	/**
+	 * A badge presenting a progress indicator
+	 */
+	export interface ProgressBadge {
+	}
```

Comment:
This may require a boolean property so that the type can be differentiated from the other Badge types: `isInProgress: boolean`